Docs: Adds a page for enums (#18590)
supabase/supabase: apps/docs/pages/guides/database/postgres/enums.mdx

## Using Enums in tables

To use the Enum in a table, you can define a column with the Enum type. For example:

{/* prettier-ignore */}
```sql
create table person (
  id serial primary key,
  name text,
  current_mood mood
);
```

Here, the `current_mood` column can only have values from the "mood" Enum.

### Inserting Data with Enums

You can insert data into a table with Enum columns by specifying one of the Enum values:

{/* prettier-ignore */}
```sql
insert into person
  (name, current_mood)
values
  ('Alice', 'happy');
```

### Querying Data with Enums

When querying data, you can filter and compare Enum values as usual:

{/* prettier-ignore */}
```sql
select * 
from person 
where current_mood = 'sad';
```

## Managing Enums

You can manage your Enums using the `alter type` statement. Here are some examples:

### Updating Enum Values

You can update the value of an Enum column:

{/* prettier-ignore */}
```sql
update person
set current_mood = 'excited'
where name = 'Alice';
```

### Adding Enum Values

> To add new values to an existing Postgres Enum, you can use the `ALTER TYPE` statement. Here's how you can do it:

Reviewer: ```suggestion
To add new values to an existing Postgres Enum, you can use the `ALTER TYPE` statement.
```

## Creating Enums

You can define a Postgres Enum using the `create type` statement. Here's an example:

{/* prettier-ignore */}
```sql
create type mood as enum (
  'happy',
  'sad',
  'excited',
  'calm'
);
```

In this example, we've created an Enum called "mood" with four possible values.

## Using Enums in tables

To use the Enum in a table, you can define a column with the Enum type. For example:

{/* prettier-ignore */}
```sql
create table person (
  id serial primary key,
  name text,
  current_mood mood
);
```

Here, the `current_mood` column can only have values from the "mood" Enum.

### Inserting Data with Enums

You can insert data into a table with Enum columns by specifying one of the Enum values:

{/* prettier-ignore */}
```sql
insert into person
  (name, current_mood)
values
  ('Alice', 'happy');
```

### Querying Data with Enums

When querying data, you can filter and compare Enum values as usual:

{/* prettier-ignore */}
```sql
select * 
from person 
where current_mood = 'sad';
```

## Managing Enums

You can manage your Enums using the `alter type` statement. Here are some examples:

> ## Updating Enum Values

Reviewer 2: ```suggestion
### Updating Enum Values
```

Should be one level lower?

You can update the value of an Enum column:

{/* prettier-ignore */}
```sql
update person
set current_mood = 'excited'
where name = 'Alice';
```

### Adding Enum Values

To add new values to an existing Postgres Enum, you can use the `ALTER TYPE` statement. Here's how you can do it:

Let's say you have an existing Enum called `mood`, and you want to add a new value, `content`:

{/* prettier-ignore */}
```sql
alter type mood add value 'content';
```

### Removing Enum Values

You can remove values from an existing Enum using the `ALTER TYPE` statement:

{/* prettier-ignore */}
```sql

> alter type mood add value 'content';

Reviewer 2: Query is for adding a value, but it should be the query to remove

```

### Getting a list of Enum Values

Check your existing Enum values by querying the enum_range function:

{/* prettier-ignore */}
```sql
select enum_range(null::mood);
```

## Reordering Enum Values

> Once an Enum is in use, it's difficult to make changes the the order of the values. However, you can do it by dropping and recreating the Enum. Here's how you can do it for the `mood` Enum above, reodering the values to be in alphabetical order.

Reviewer 2: ```suggestion
Once an Enum is in use, it's difficult to make changes to the order of the values. However, you can do it by dropping and recreating the Enum. Here's how you can do it for the `mood` Enum above, reodering the values to be in alphabetical order.
```

import Layout from '~/layouts/DefaultGuideLayout'

export const meta = {
  id: 'enums',
  title: 'Managing Enums in Postgres',

>   description: 'Define a strict set of values for application.',

Reviewer: ```suggestion
  description: 'Define a strict set of values that can be used in table and function definitions.',
```

Nit: perhaps more specific?
